### ISSUE TITLE
Multi-dimensional list types

### ISSUE THREAD

**Issue by tunurgitr** [ChilliCream/hotchocolate#676]
**Is your feature request related to a problem? Please describe.**
I have a graph that uses a multi-dimensional list (`[[String!]!]!`) Multi-dimensional lists are explicitly not supported https://github.com/ChilliCream/hotchocolate/blob/master/src/Core/Types/Types/ListType.cs#L28-L30 but aren't specifically prohibited in the graphql spec.

Especially in the context of stitching, we don't always have full control over the schema we're consuming, so matching the spec is critical.

**Describe the solution you'd like**
The implementation to match the graphql spec.

I looked through other issues and I wasn't able to determine if this is already planned or if there is any kind of significant roadblock to implementing this.

**Comment by michaelstaib** [comment]
Hi @tunurgitr,

we have that one on our todo list. We will fix this with the next version. I am sorry that you ran into this issue but when we conceived HC we postponed this one since it makes a lot things more complicated. With the new type system almost read for a preview build we will fix this flaw in our implementation very soon. I hope to get this done in the next three weeks.

**Comment by michaelstaib** [comment]
#369

**Comment by michaelstaib** [comment]
https://graphql.github.io/graphql-spec/June2018/#sec-Type-System.List

List Input Coercion Rules for Lists

Expected Type | Provided Value | Coerced Value
-- | -- | --
[[Int]] | [[1], [2, 3]] | [[1], [2, 3]
[[Int]] | [1, 2, 3] | Error: Incorrect item value
[[Int]] | 1 | [[1]]
[[Int]] | null | null

**Comment by michaelstaib** [comment]
@tunurgitr we will start on this one now.

**Comment by michaelstaib** [comment]
Just to give an update on this ... we have added a lot of smaller fixes fir this. We are not yet switching the button, but we have updated most of the algorithms handling types. There are still some the expect a fixed four level type structure especially when we infer types but this one is being worked in. 

The client especially brought a ton of fixes for that .... I expect this one to be fixed in the V11 time frame.
